### ISSUE TITLE
Compilation unrecognized exception with default String? = None

### ISSUE THREAD

**Issue by Ben-Habermeyer** [dnanexus/dxCompiler#351]
Hi DNAnexus, I receive a compilation error when using a workflow with an optional string default on dxCopmiler 2.10.2 
```
workflow2 {
  input {
    Int arg1 = 1
    String? caller_to_emulate = None
  }
}
```
and invoked without providing caller_to_emulate
```
workflow1 {
...
call workflow2 {
    input:
      arg1=1
  }
}
```

There are a bunch of lines like `[warning] unable to parse field caller_to_emulate default value null` in the verbose trace. The exact exception raised seems to be 

```
[error] java.lang.Exception: Unrecognized expression ValueNull(T_Any)
	at wdlTools.generators.code.WdlGenerator.wdlTools$generators$code$WdlGenerator$$buildExpression(WdlGenerator.scala:825)
	at wdlTools.generators.code.WdlGenerator$DeclarationStatement.$anonfun$rhs$1(WdlGenerator.scala:944)
	at scala.Option.map(Option.scala:242)
	at wdlTools.generators.code.WdlGenerator$DeclarationStatement.<init>(WdlGenerator.scala:942)
	at wdlTools.generators.code.WdlGenerator$InputsBlock.$anonfun$body$5(WdlGenerator.scala:992)
	at scala.collection.immutable.Vector1.map(Vector.scala:1886)
	at scala.collection.immutable.Vector1.map(Vector.scala:375)
	at wdlTools.generators.code.WdlGenerator$InputsBlock.body(WdlGenerator.scala:989)
	at wdlTools.generators.code.WdlGenerator$BlockStatement.<init>(WdlGenerator.scala:964)
	at wdlTools.generators.code.WdlGenerator$InputsBlock.<init>(WdlGenerator.scala:987)
	at wdlTools.generators.code.WdlGenerator$TaskBlock.body(WdlGenerator.scala:1336)
	at wdlTools.generators.code.WdlGenerator$BlockStatement.<init>(WdlGenerator.scala:964)
	at wdlTools.generators.code.WdlGenerator$TaskBlock.<init>(WdlGenerator.scala:1329)
	at wdlTools.generators.code.WdlGenerator$DocumentSections$$anonfun$format$4.applyOrElse(WdlGenerator.scala:1396)
	at wdlTools.generators.code.WdlGenerator$DocumentSections$$anonfun$format$4.applyOrElse(WdlGenerator.scala:1395)
	at scala.collection.StrictOptimizedIterableOps.collect(StrictOptimizedIterableOps.scala:151)
	at scala.collection.StrictOptimizedIterableOps.collect$(StrictOptimizedIterableOps.scala:136)
	at scala.collection.immutable.Vector.collect(Vector.scala:113)
	at wdlTools.generators.code.WdlGenerator$DocumentSections.format(WdlGenerator.scala:1395)
	at wdlTools.generators.code.WdlGenerator.generateElement(WdlGenerator.scala:1420)
	at wdlTools.generators.code.WdlGenerator.generateDocument(WdlGenerator.scala:1428)
	at dx.core.languages.wdl.VersionSupport.generateDocument(VersionSupport.scala:93)
	at dx.core.languages.wdl.WdlDocumentSource.toString(WdlCodeGenerator.scala:16)
	at dx.compiler.ApplicationCompiler.apply(ApplicationCompiler.scala:618)
	at dx.compiler.Compiler$BundleCompiler.maybeBuildApplet(Compiler.scala:358)
	at dx.compiler.Compiler$BundleCompiler.$anonfun$apply$1(Compiler.scala:472)
	at scala.collection.IterableOnceOps.foldLeft(IterableOnce.scala:646)
	at scala.collection.IterableOnceOps.foldLeft$(IterableOnce.scala:642)
	at scala.collection.AbstractIterable.foldLeft(Iterable.scala:926)
	at dx.compiler.Compiler$BundleCompiler.apply(Compiler.scala:442)
	at dx.compiler.Compiler.apply(Compiler.scala:495)
	at dxCompiler.Main$.compile(Main.scala:541)
	at dxCompiler.Main$.dispatchCommand(Main.scala:794)
	at dxCompiler.Main$.main(Main.scala:924)
	at dxCompiler.MainApp$.delayedEndpoint$dxCompiler$MainApp$1(Main.scala:929)
	at dxCompiler.MainApp$delayedInit$body.apply(Main.scala:928)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1(App.scala:76)
	at scala.App.$anonfun$main$1$adapted(App.scala:76)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:926)
	at scala.App.main(App.scala:76)
	at scala.App.main$(App.scala:74)
	at dxCompiler.MainApp$.main(Main.scala:928)
	at dxCompiler.MainApp.main(Main.scala)
```

Removing the `None` to just `String? caller_to_emulate` resolves the issue, but I'm not sure why that'd be the case.

**Comment by sclan** [comment]
I don't have an answer to this but I do observe the same line
String? caller_to_emulate = None
works if it is placed outside of the input block (private declaration) and with "version 1.1" header.

**Comment by emiloslavsky** [comment]
Will take a look (dnanexus tracker app-1222)

**Comment by Gvaihir** [comment]
@Ben-Habermeyer What's the WDL version?
I can reproduce with 1.1 and all default flags for compiler.
Could you confirm the WDL version and the flags for the compile command?

**Comment by Gvaihir** [comment]
When compiling in the `-locked` mode - no exception thrown. Fixed for default "unlocked". Expect in the next release.

**Comment by Ben-Habermeyer** [comment]
> @Ben-Habermeyer What's the WDL version? I can reproduce with 1.1 and all default flags for compiler. Could you confirm the WDL version and the flags for the compile command?

Yes we are using dxCompiler latest version with WDL version 1.1 (I think you need 1.1 to declare `None`). We use `-archive -extras -defaults -projectWideReuse`

**Comment by Ben-Habermeyer** [comment]
Was this ever released? Still seeing this exception with `Float? = None` as well

**Comment by Gvaihir** [comment]
Not yet. Sorry for the delay - we are wrapping up other changes too.

**Comment by Gvaihir** [comment]
Fixed in [2.10.3](https://github.com/dnanexus/dxCompiler/releases/tag/2.10.3)
